### ISSUE TITLE
oas3-valid-schema-example does not support non-string enums

### ISSUE THREAD

**Issue by rliebz** [daveshanley/vacuum#229]
Vacuum appears to use treat all `enum` values as strings when validating examples, regardless of their actual type.

### Example Config

```yaml
components:
  schemas:
    NotAuthorized:
      type: integer
      enum: [401]
      example: 401
```

### Expected
No error pertaining to the example, as it is a member of the enum and the specified type

### Actual 
I get an error for `oas3-valid-schema-example` with the following message:
>  Example for component `NotAuthorized` is not valid: `(root) must be one of the following: "401"`. Value `401` is not compatible

**Comment by daveshanley** [comment]
Interesting, I will investigate this - it may be a dependency issue.

**Comment by daveshanley** [comment]
github.com/xeipuuv/gojsonschema is throwing this error.

`vaccum` also uses github.com/santhosh-tekuri/jsonschema, so I will see if that handles the schema correctly.

**Comment by daveshanley** [comment]
Resolved in `v0.1.0`
